tfhe/src/integer/server_key/radix/mod.rs

```diff
@@ -157,8 +157,18 @@ impl ServerKey {
     /// assert_eq!(msg + msg, res);
     /// ```
     pub fn full_propagate<PBSOrder: PBSOrderMarker>(&self, ctxt: &mut RadixCiphertext<PBSOrder>) {
+        self.partial_propagate(ctxt, 0);
+    }
+
+    /// Propagates all carries except the last one.
+    /// For development purpose only.
+    fn partial_propagate<PBSOrder: PBSOrderMarker>(
+        &self,
+        ctxt: &mut RadixCiphertext<PBSOrder>,
+        start_index: usize,
+    ) {
         let len = ctxt.blocks.len();
-        for i in 0..len {
+        for i in start_index..len {
             self.propagate(ctxt, i);
         }
     }
```

Comment:
is partial propagate really dev only then ?

Reply:
I've left the comment as it was originally, but yes its not really a dev only (as in you use it to debug) function

Reply:
changed the comment